lib/facter/util/linux/socket_parser.rb

```diff
@@ -59,8 +59,23 @@ def mac_from_bonded_interface(interface_name)
           end
 
           def bond_master_of(interface_name)
-            content = Facter::Core::Execution.execute("ip link show #{interface_name}", logger: @log)
-            content.match(/master (\S*) /)&.captures&.first
+            content = get_ip_link_show_data(interface_name)
+            content&.match(/master (\S*) /)&.captures&.first
+          end
+
+          def get_ip_link_show_data(interface_name)
+            @ip_link_show_data ||= read_ip_link_show_data
+            @ip_link_show_data[interface_name]
+          end
+
+          def read_ip_link_show_data
+            ip_link_show_data = {}
+            output = Facter::Core::Execution.execute('ip -o link show', logger: @log)
+            output.each_line do |line|
+              interface_name = line&.split(':')&.[](1)&.strip
```

Comment:
Could be more readable like: `line.split(':')[1].strip if line`

Reply:
We need to make sure the code is not raising, maybe `line.split(':')[1]&.strip if line`?

Reply:
Good point, didn't consider that, your suggestion sounds good then 👍🏼